.github/workflows/deploy-to-environment.yml

```diff
@@ -42,8 +43,42 @@ jobs:
           GIT_BRANCH=${GIT_REF##*/}
           INPUT=${{ github.event.inputs.environment }}
           ENVIRONMENT=${INPUT:-"development"}
+          RELEASE=${ENVIRONMENT,,}-`date +%Y-%m-%d`.${{ github.run_number }}
           echo "environment=${ENVIRONMENT,,}" >> $GITHUB_OUTPUT
           echo "branch=$GIT_BRANCH" >> $GITHUB_OUTPUT
+          echo "release=${RELEASE,,}" >> $GITHUB_OUTPUT
+
+  create-tag:
+    if: needs.set-env.outputs.release == 'production'
```

Comment:
The value of `needs.set-env.outputs.release` is in the format `<environment>-YYYY-MM-DD.X` where `<environment>` is `development`,`test`, or `production` and `X` is the iteration count of this workflow (auto incrementing number).

As a result, your `if` statement here will never evaluate as true. If you want to check what environment it is you should change `outputs.release` to `outputs.environment` :)